packages/flutter/lib/src/material/dialog.dart

```diff
@@ -634,10 +703,73 @@ class AlertDialog extends StatelessWidget {
   /// button bar.
   final bool scrollable;
 
+  /// A scroll controller that can be used to control the scrolling of the
+  /// [content] in the dialog.
+  ///
+  /// Defaults to null, and is typically not needed, since most alert messages
+  /// are short.
+  ///
+  /// Only used when [AlertDialog.adaptive] resolves to an iOS platform.
+  ///
+  /// See also:
+  ///
+  ///  * [actionScrollController], which can be used for controlling the actions
+  ///    section when there are many actions.
+  final ScrollController? scrollController;
```

Comment:
I'm considering if we should add these 4 APIs here because the material AlertDialog will not use them at all. Another way I can think of is to use `factory` keyword when we create `AlertDialog.adaptive` constructor, and the factory constructor will be assigned to a private subclass which has the extra APIs. An example can be `FilledButton.icon()`. By using factory, we can add new parameters only in the `AlertDialog.adaptive` constructor without adding new fields for `AlertDialog`. I think that would be something like: https://gist.github.com/QuncCccccc/93f22070b6b459912eeddb03ff55d1ec

But please let me know if this doesn't make sense:)

Reply:
No that makes sense, and that looks a lot cleaner. Thank you!

Reply:
that's way better.